tests/CommonExtractor.php

```diff
@@ -0,0 +1,497 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Tests;
+
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvWriter;
+use Keboola\Csv\Exception as CsvException;
+use Keboola\DbExtractorCommon\Configuration\BaseExtractorConfig;
+use Keboola\DbExtractorCommon\Configuration\DatabaseParameters;
+use Keboola\DbExtractorCommon\Configuration\TableDetailParameters;
+use Keboola\DbExtractorCommon\Configuration\TableParameters;
+use Keboola\DbExtractorCommon\Exception\ApplicationException;
+use Keboola\DbExtractorCommon\Exception\DeadConnectionException;
+use Keboola\DbExtractorCommon\BaseExtractor;
+use Keboola\DbExtractorCommon\RetryProxy;
+
+class CommonExtractor extends BaseExtractor
+{
+    public const TYPE_AUTO_INCREMENT = 'autoIncrement';
+    public const TYPE_TIMESTAMP = 'timestamp';
+
+    /** @var \PDO|null */
+    private $connection;
+
+    /** @var array */
+    private $incrementalFetching;
+
+    public function extract(BaseExtractorConfig $config): array
+    {
+        $imported = [];
+        $outputState = [];
+        if ($config->isConfigRow()) {
+            $tableParameters = TableParameters::fromRaw($config->getParameters());
+            $exportResults = $this->extractTable($tableParameters);
+            if (isset($exportResults['state'])) {
+                $outputState = $exportResults['state'];
+                unset($exportResults['state']);
+            }
+            $imported = $exportResults;
+        } else {
+            foreach ($config->getEnabledTables() as $table) {
+                $exportResults = $this->extractTable($table);
+                $imported[] = $exportResults;
+            }
+        }
+
+        return [
```

Comment:
Tady přemýšlím, jestli to musí být takhle - co když udělám extractor, který nevrátí status success? Jak vím, že má mít status success? A jak vůbec mám vědět, že mám vrátit tyhle tři klíče zrovna? Co na to udělat nějaký value object? (ptám se fakt protože si nejsem jistý - @odinuv @pivnicek @Actimel )

Reply:
Nemyslím v téhle konkrétní implementaci, ale aby ho poskytoval právě ten common

Reply:
Dulezite je jen to pole state, jinak tam muze byt cokoli. Ale zaslouzilo by si to nejaky hezci tvar. Nechal bych to na jine issue.